security/advancedtls/crl.go

```diff
@@ -516,3 +516,34 @@ func extractCRLIssuer(crlBytes []byte) ([]byte, error) {
 	}
 	return issuer, nil
 }
+
+func hasExpired(crl x509.RevocationList, now time.Time) bool {
+	return !now.Before(crl.NextUpdate)
+}
+
+// pemCRLPrefix, pemType, and parseRevocationList  come largely from here
+// x509.go:
+// https://github.com/golang/go/blob/e2f413402527505144beea443078649380e0c545/src/crypto/x509/x509.go#L1669-L1690
+// We must first convert PEM to DER to be able to use the new
+// x509.ParseRevocationList instead of the deprecated x509.ParseCRL
+
+// pemCRLPrefix is the magic string that indicates that we have a PEM encoded
+// CRL.
+var pemCRLPrefix = []byte("-----BEGIN X509 CRL")
```

Comment:
This is a duplicate of line 495. The pemType below is also used in extractCRLIssuer() locally. In fact, the logic in the prefix check of that function and the new one here is the same so I think it's worth refactoring it out as a helper func.

Reply:
Removed this and functionalized the conversion